static/js/compose.js

```diff
@@ -414,7 +415,10 @@ export function initialize() {
     ).on("keyup", update_on_recipient_change);
     $(
         "#stream_message_recipient_stream,#stream_message_recipient_topic,#private_message_recipient",
-    ).on("change", update_on_recipient_change);
+    ).on("change", () => {
+        update_on_recipient_change();
+        compose_state.set_recipient_edited_manually(true);
+    });
```

Comment:
Does this event handler get called when the recipient changes due to automated changes, say, clicking on a private message conversation (which opens the compose box)? I worry a bit that we might end up setting the "recipient edited manually" property in cases where it shouldn't be.

Reply:
(I would consider doing some testing with this with a bonus commit where we make this handler trigger a highly visible side effect, like changing the compose box border to be red, to confirm things work as expected).

Reply:
This event is only triggered when the user manually and directly changes the content in the recipient fields. Have now also tested with a red outline.

Reply:
Awesome, thanks for verifying!